lib/Sub/Future.pm

```diff
@@ -53,21 +53,24 @@ until the value is ready.
 
 sub future(&) {
     my ($code) = @_;
-    
+    my ($pid, $reader, $writer);
+
+    # open bidirectional pipe
+    pipe($reader, $writer) || die "Can't open pipe: $!\n";
+
     # create a new process to handle the spawned calculation
-    my $pid = open my $fh, '-|';
+    $pid = fork();
```

Comment:
Please leave this variable declaration here. In general, the code declares variables as close to their first assignment as possible.